sycl/test-e2e/Matrix/element_wise_all_sizes_impl.hpp

```diff
@@ -0,0 +1,129 @@
+static constexpr size_t M_MULTIPLIER = 16;
+
+static float make_fp32(bfloat16 x) {
+  unsigned int y = *((int *)&x);
+  y = y << 16;
+  float *res = reinterpret_cast<float *>(&y);
+  return *res;
+}
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+template <typename T, typename RefT, size_t M, size_t N>
+void assert_ops_ref(host_accessor<T, 2, access::mode::read_write> C,
+                    const RefT ref) {
+  for (size_t i = 0; i < M; i++)
+    for (size_t j = 0; j < N; j++) {
+      if (std::is_same_v<RefT, float>) {
+        auto diff = make_fp32(C[i][j]) - ref;
+        assert(std::fabs(static_cast<float>(diff)) <
+               std::numeric_limits<float>::epsilon());
+      } else if (std::is_same_v<RefT, int>) {
+        assert(C[i][j] == ref);
+      }
+    }
+}
+
+template <typename T, typename RefT, size_t M, size_t K, size_t TM, size_t TK>
+void matrix_verify_add(queue q, big_matrix<T, M, K> &A, nd_range<2> &r,
+                       const RefT ref) {
+  buffer<T, 2> bufA(A.get_data(), range<2>(M, K));
+
+  q.submit([&](handler &cgh) {
+     sycl::accessor accA{bufA, cgh, sycl::read_write};
+
+     cgh.parallel_for(r, [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(
+                             SG_SZ)]] {
+       const auto global_idx = spmd_item.get_global_id(0);
+       const auto global_idy = spmd_item.get_global_id(1);
+       const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+       const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+       sub_group sg = spmd_item.get_sub_group();
+       joint_matrix<sub_group, T, use::a, TM, TK, layout::row_major> sub_a;
+
+       if constexpr (std::is_same_v<T, bfloat16>) {
```

Comment:
if 5 and 2 are passed as parameters of the correct type in add_ref as suggested below, then I guess no need to keep this condition here, code can be simplified like:
joint_matrix_fill(sg, sub_a, param1);

Reply:
Done